crates/sui-core/src/transaction_outputs.rs

```diff
@@ -94,7 +93,7 @@ impl TransactionOutputs {
             .filter_map(|(id, ((version, digest), owner))| {
                 owner.is_address_owned().then_some((id, version, digest))
             })
-            .chain(received_objects)
+            .chain(received_objects.into_iter())
```

Comment:
Clippy says we don't need this.
```suggestion
            .chain(received_objects)
```